picframe/viewer_display.py

```diff
@@ -224,11 +257,13 @@ def __tex_load(self, pics, size=None):
                 im = im.resize((max_dimension, int(h * max_dimension / w)), resample=Image.BICUBIC)
             elif h > max_dimension:
                 im = im.resize((int(w * max_dimension / h), max_dimension), resample=Image.BICUBIC)
-            if self.__blur_edges and size is not None:
-                wh_rat = (size[0] * im.size[1]) / (size[1] * im.size[0])
-                if abs(wh_rat - 1.0) > 0.01: # make a blurred background
+
+            screen_aspect, image_aspect, diff_aspect = self.__get_aspect_diff(size, im.size)
+
+            if self.__blur_edges and size:
+                if diff_aspect > 0.01:
```

Comment:
@paddywwoof - You might check this out and see if you agree with my replacement for the original `wh_rat` code...